### PR TITLE
merge istio.io/pkg to istio.io/istio/pkg

### PR DESCRIPTION
**Pull request by zirain**
**Please provide a description of this PR:**

related to https://github.com/istio/istio/issues/44991

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -145,7 +145,7 @@ require (
 	github.com/go-jose/go-jose/v3 v3.0.0 // indirect
 	github.com/go-kit/log v0.2.1 // indirect
 	github.com/go-logfmt/logfmt v0.6.0 // indirect
-	github.com/go-logr/logr v1.2.4 // indirect
+	github.com/go-logr/logr v1.2.4
```
Comment (jacob-delgado) [inline go.mod:148]: can we move all indirect modules to the other require directive?

**Review by howardjohn — COMMENTED** [review]
I opened https://github.com/istio/pkg/pull/813. Can move to this PR if its easier (or after this PR)

LGTM but want someone else to approve as well, as its a large PR.

File: go.mod (hunk 1 above)
Comment (ericvn) [reply go.mod:148]: Surprised a `go mod tidy` didn't pick this up.

**Review by hanxiaop — APPROVED** [review]
lgtm

File: go.mod (hunk 1 above)
Comment (zirain) [reply go.mod:147]: yes, we can revise this later.

**Review by hzxuzhonghu — APPROVED** [review]

**Review by ericvn — APPROVED** [review]

File: prow/release-commit.sh [hunk 2]
```diff
@@ -66,7 +66,7 @@ ${DEPENDENCIES:-$(cat <<EOD
     auto: deps
   pkg:
     git: https://github.com/istio/pkg
-    auto: modules
+    branch: master
```
Comment (howardjohn) [inline prow/release-commit.sh:69]: Should remove `pkg` entirely (L67-69). We will need the same change in istio/release-builder as a followup

**Review by howardjohn — APPROVED** [review]

**Review by lei-tang — APPROVED** [review]

**Review by jacob-delgado — APPROVED** [review]

**Pull request merged**